shared/js/dialer/keypad.js

```diff
@@ -504,6 +516,139 @@ var KeypadManager = {
     }
   },
 
+  /**
+   * Returns true if the number is a speed dial code as described in
+   * 3GPP TS 22.030 6.6.4. Speed dial codes are in the N(N)(N)# format.
+   */
+  _isSpeedDialNumber: function(number) {
+    return !!number.match(/^[0-9][0-9]{0,2}\#$/);
+  },
```

Comment:
nit: Should we be matching on 0# ?  I'm not sure if our export function is wrong or the spec is wrong or the code is wrong.  I saw one note in the code that the speed dials are 1 indexed.  (As an aside, I started with an empty SIM contacts and exported one contact and it saved it as 1#.)